OpenRA.Mods.Common/Traits/AutoTarget.cs

```diff
@@ -190,7 +190,7 @@ Actor ChooseTarget(Actor self, WDist range, bool allowMove)
 								&& !target.IsInRange(self.CenterPosition, arm.Weapon.MinRange))), a);
 				})
 
-				.Where(kv => kv.Key != null && self.Owner.CanViewActor(kv.Value))
+				.Where(kv => kv.Key != null && (self.Owner.HasFogVisibility || self.Owner.CanViewActor(kv.Value)))
```

Comment:
Use `Player.CanTargetActor` instead - that already includes this check and is more appropriate for use by `AutoTarget`.